Kudu.Core/Deployment/Generator/ExternalCommandBuilder.cs

```diff
@@ -85,5 +114,94 @@ protected void RunCommand(DeploymentContext context, string command)
+            {
+                defaultPath = System.Environment.GetEnvironmentVariable(WellKnownEnvironmentVariables.PostDeploymentActionsDirectoryKey);
+            }
+
+            if (FileSystemHelpers.DirectoryExists(defaultPath))
+            {
+                scriptFolders.Add(defaultPath);
+            }
+
+            // D:\Program Files (x86)\SiteExtensions
+            string siteExtensionFolder = Path.Combine(System.Environment.GetFolderPath(System.Environment.SpecialFolder.ProgramFilesX86), "SiteExtensions");
+            // Find all enviroment variable with format {package id}_EXTENSION_VERSION,
+            // And find scripts from PostDeploymentActions" folder
+            foreach (DictionaryEntry entry in System.Environment.GetEnvironmentVariables())
+            {
+                string key = entry.Key.ToString();
+                string val = entry.Value.ToString();
+                NuGetVersion ver = null;
+                if (key.EndsWith("_EXTENSION_VERSION", StringComparison.Ordinal)
+                    && (string.Equals("latest", val, StringComparison.OrdinalIgnoreCase) || NuGetVersion.TryParse(val, out ver)))   // either latest or a specific version
+                {
+                    if (ver != null)
+                    {
+                        // "D:\Program Files (x86)\SiteExtensions", "{PackageName}", "{version}", "PostDeploymentActions"
+                        //  => D:\Program Files (x86)\SiteExtensions\{PackageName}\{version}\PostDeploymentActions
+                        string scriptFolder = Path.Combine(siteExtensionFolder, key.Replace("_EXTENSION_VERSION", string.Empty), ver.ToNormalizedString(), PostDeploymentActions);
+                        if (FileSystemHelpers.DirectoryExists(scriptFolder))
+                        {
+                            scriptFolders.Add(scriptFolder);
+                        }
+                    }
+                    else
+                    {
+                        // Get the latest version
+                        // "D:\Program Files (x86)\SiteExtensions", "{PackageName}"
+                        //  => D:\Program Files (x86)\SiteExtensions\{PackageName}
+                        string packageRoot = Path.Combine(siteExtensionFolder, key.Replace("_EXTENSION_VERSION", string.Empty));
+                        if (FileSystemHelpers.DirectoryExists(packageRoot))
+                        {
+                            string[] allVersions = FileSystemHelpers.GetDirectories(packageRoot);   // expecting a list of version e.g D:\Program Files (x86)\SiteExtensions\{PackageName}\{version}
+                            if (allVersions.Count() > 0)
+                            {
+                                // latest version comes first
+                                Array.Sort(allVersions, (string v1, string v2) =>
+                                {
+                                    NuGetVersion ver1 = NuGetVersion.Parse(new DirectoryInfo(v1).Name);
```

Comment:
we will see below message from stack trace. should be good enough for us to identify the issue : ```` ---- System.ArgumentException : 'blabla' is not a valid version string. ````

````
Result1 StackTrace:	
at System.Collections.Generic.GenericArraySortHelper`1.Sort(T[] keys, Int32 index, Int32 length, IComparer`1 comparer)
   at System.Array.Sort[T](T[] array, Int32 index, Int32 length, IComparer`1 comparer)
   at System.Array.Sort[T](T[] array, IComparer`1 comparer)
   at System.Array.Sort[T](T[] array, Comparison`1 comparison)
   at Kudu.Core.Deployment.Generator.ExternalCommandBuilder.GetPostBuildActionScripts() in E:\kudu\Kudu.Core\Deployment\Generator\ExternalCommandBuilder.cs:line 174
   at Kudu.Core.Test.Deployment.Generator.ExternalCommandBuilderFacts.GetPostBuildActionScriptsShouldReturnActionScriptFromSiteExtensions() in E:\kudu\Kudu.Core.Test\Deployment\Generator\ExternalCommandBuilderFacts.cs:line 171
----- Inner Stack Trace -----
   at NuGet.Versioning.NuGetVersion.Parse(String value)
   at Kudu.Core.Deployment.Generator.ExternalCommandBuilder.<>c.<GetPostBuildActionScripts>b__27_0(String v1, String v2) in E:\kudu\Kudu.Core\Deployment\Generator\ExternalCommandBuilder.cs:line 176
   at System.Array.FunctorComparer`1.Compare(T x, T y)
   at System.Collections.Generic.ArraySortHelper`1.InsertionSort(T[] keys, Int32 lo, Int32 hi, IComparer`1 comparer)
   at System.Collections.Generic.ArraySortHelper`1.IntroSort(T[] keys, Int32 lo, Int32 hi, Int32 depthLimit, IComparer`1 comparer)
   at System.Collections.Generic.ArraySortHelper`1.IntrospectiveSort(T[] keys, Int32 left, Int32 length, IComparer`1 comparer)
   at System.Collections.Generic.GenericArraySortHelper`1.Sort(T[] keys, Int32 index, Int32 length, IComparer`1 comparer)
Result1 Message:	
System.InvalidOperationException : Failed to compare two elements in the array.
---- System.ArgumentException : 'blabla' is not a valid version string.
Parameter name: value
````